docs/functional-areas/p2p_DTR_api.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Token Ring-P2P API
+page_id: DTR_p2p_api
+---
+
+## Introduction
+An extra layer of communication is added to the Crazyflie API to allow for robust and more reliable peer to peer communication. This is done by implementing a token ring protocol on top of the peer to peer API.The Token Ring Protocol is assuming that the nodes are static and the communication is not changing. The protocol is implemented by using a token ring structure. The token ring structure is a circular linked list of nodes. The nodes are connected by a ring of links. The ring is formed by the nodes in the order they are connected. The first node is connected to the last node and the last node is connected to the first node. This protocol is used to ensure that each time only one Crazyflie broadcasts data which leads to less packet collisions and losses. It also provides a way to ensure that the transmitted data will be sent to the other copters since it receives an acknowledgement from each receiver.
+
+Currently, peer to peer communication on the Crazyflie along with Token Ring Protocol are **in development**. An API is made available to send and receive packets to and from other Crazyflies by letting the protocol automatically handle the lower level interactions with P2P API. The protocol runs as a separate task and utilizes a separate port of the P2P API. Thus the user is given the freedom to use both the P2P API and the DTR protocol at the same time,depending on the use case and the problem to be solved.
+
+The interface with the Token Ring Protocol is achieved by using 2 queues. One queue is used to send messages to the protocol and the other queue is used to receive messages from the protocol. In that sense ,the execution of the protocol won't block the execution of the rest of the user code and the user can send and receive messages asynchronously.
+
+
+## Using the Token Ring API
+Functions and structures are defined in the header files `src/modules/interface/p2pDTR/DTR_types.h` and `src/modules/interface/p2pDTR/token_ring.h`. There is also an app layer example (`/app_p2p_DTR/`) available in the example folder of the repository.
+
+Each packet has the following structure:
+
+``` C
+typedef struct radio_packet {
+	uint8_t packetSize;
+	uint8_t message_type;
+	uint8_t source_id;
+	uint8_t target_id;
+	uint8_t dataSize;
+	bool allToAllFlag;
```

Comment:
Another option to using the `allToAllFlag` would be to reserve target_id == 0xff for this purpose.

Reply:
fixed that on later commit